### PR TITLE
Update Filecoin RPC to be compatible with Lotus RemoteWallet

### PR DESCRIPTION
**Pull request by jframe**
Update the Filecoin RPC so that it works when called using the Lotus RemoteWallet as a signer.

* Update the path to be /rpc/v0 instead of /rpc/v1/filecoin as this is hard-coded in Lotus
* Add support for the MsgMeta field in the Wallet.Sign
* Verify that the Wallet.Sign is valid by computing the CID(MsgMeta.extra) == data
* Remove the unused Wallet.SignMessage RPC

### CODE REVIEW

**Review by usmansaleem — APPROVED** [review]
LGTM

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class FcCidEncoder {
+
+  private static final BigInteger FC_HASHING_ALGO_CODE = BigInteger.valueOf(0xb220);
+  private static final byte CID_VERSION = (byte) 1;
+  private static final byte DagCBOR_CODEC_ID = (byte) 113;
+
+  public Bytes createCid(final Bytes cborEncodedBytes) {
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java:28]: do we care if its cborencodedbytes - or is simply data?

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import tech.pegasys.web3signer.core.util.Blake2b;
+import tech.pegasys.web3signer.core.util.ByteUtils;
+
+import java.math.BigInteger;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class FcCidEncoder {
+
+  private static final BigInteger FC_HASHING_ALGO_CODE = BigInteger.valueOf(0xb220);
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java:24]: a comment that these are hard-coded magic numbers from lotus?

**Review by rain-on — APPROVED** [review]

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java (hunk 2 above)
Comment (jframe) [reply core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcCidEncoder.java]: Replacing all these constants and using the Java CID library. That should make all of this clearer.

**Pull request merged**